plugins/modules/s3_bucket.py

```diff
@@ -585,7 +585,7 @@ def get_current_bucket_tags_dict(s3_client, bucket_name):
     try:
         current_tags = s3_client.get_bucket_tagging(Bucket=bucket_name).get('TagSet')
     except ClientError as e:
-        if e.response['Error']['Code'] == 'NoSuchTagSet':
+        if e.response['Error']['Code'] == 'NoSuchTagSetError':
```

Comment:
While I understand that most of this module uses the `if e.response['Error']['Code']` mechanism for catching specific AWS errors (and I certainly do **not** expect you to correct all of them).  We now have a nicer mechanism for limiting which Exceptions we catch, and allowing the remaining ones to propagate without needing to re-raise them.

This function could be nicely simplified to:
```
def get_current_bucket_tags_dict(s3_client, bucket_name):
    try:
        current_tags = s3_client.get_bucket_tagging(Bucket=bucket_name).get('TagSet')
    except is_boto3_error_code('NoSuchTagSetError') as e:
        return {}
```

Would you mind switching your patch over to this version please?